app/controllers/api/v2/courses_controller.rb

```diff
@@ -75,6 +75,7 @@ def publish
           @course.publish_sites
           @course.publish_enrichment(@current_user)
           @course.reload
+          Courses::PublishCourseService.new.execute(course: @course)
```

Comment:
The second `Course` is redundant here. Could we use `Course::PublishService` or `CourseService::Publish`? Also we should decide a consistent pattern for this sort of service. I prefer using a static `.call` as in [app/services/course_search_service.rb](app/services/course_search_service.rb) as it makes the `.new` method private. We can move all of the boiler plate in that service into a module to share.

Reply:
Can we add a card to move all of the other things into this service too once this is sufficiently ready for user research?

Reply:
> Also we should decide a consistent pattern for this sort of service. I prefer using a static `.call`

:100: